src/mixins/validatable-http-message.js

```diff
@@ -0,0 +1,450 @@
+      const expectedHeaders = {};
+      for (let name in this.expected.headers) {
+        const value = this.expected.headers[name];
+        expectedHeaders[name.toLowerCase()] = value;
+      }
+      this.expected.headers = expectedHeaders;
+    }
+  }
+
+  ensureValidationHeaders() {
+    const existingValidation = this.validation;
+    if (!existingValidation || !existingValidation.headers) {
+      this.validation = Object.assign({}, existingValidation, {
+        headers: {
+          results: []
+        }
+      });
+    }
+  }
+
+  validateHeaders() {
+    this.ensureValidationHeaders();
+    this.setHeadersRealType();
+    this.setHeadersExpectedType();
+    this.setHeadersValidator();
+    this.runHeadersValidator();
+  }
+
+  setHeadersRealType() {
+    this.ensureValidationHeaders();
+    const { headers } = this;
+    this.validation.headers.realType =
+      headers instanceof Object && !Array.isArray(headers)
+        ? 'application/vnd.apiary.http-headers+json'
+        : null;
+  }
+
+  setHeadersExpectedType() {
+    this.ensureValidationHeaders();
+    const expectedHeaders = this.expected.headers;
+    this.validation.headers.expectedType =
+      expectedHeaders instanceof Object && !Array.isArray(expectedHeaders)
+        ? 'application/vnd.apiary.http-headers+json'
+        : null;
+  }
+
+  setHeadersValidator() {
+    const expectsJson =
+      this.validation.headers.expectedType ===
+      'application/vnd.apiary.http-headers+json';
+
+    if (
+      this.validation.headers.realType ===
+        this.validation.headers.expectedType &&
+      expectsJson
+    ) {
+      return (this.validation.headers.validator = 'HeadersJsonExample');
+    } else {
+      this.validation.headers.validator = null;
+
+      if (!this.validation.headers.results) {
+        this.validation.headers.results = [];
+      }
+
+      const entry = {
+        message: `\
+No validator found for real data media type \
+"${JSON.stringify(this.validation.headers.realType)}" \
+and expected data media type \
+"${JSON.stringify(this.validation.headers.expectedType)}".\
+        `,
+        severity: 'error'
+      };
+
+      this.validation.headers.results.push(entry);
+    }
+  }
+
+  runHeadersValidator() {
+    // throw new Error JSON.stringify @validation.headers.validator, null, 2
+    let validator;
+
+    if (this.validation.headers.validator == null) {
+      this.validation.headers.rawData = null;
+    } else {
+      const ValidatorClass = validators[this.validation.headers.validator];
+      validator = new ValidatorClass(this.headers, this.expected.headers);
+      this.validation.headers.rawData = validator.validate();
+    }
+
+    if (!Array.isArray(this.validation.headers.results)) {
+      this.validation.headers.results = [];
+    }
+
+    if (this.validation.headers.rawData !== null) {
+      const results = validator.evaluateOutputToResults();
+      this.validation.headers.results = results.concat(
```

Comment:
Agree with @artem-zakharchenko. It's fishy, but let's not start fixing bugs in this change, otherwise it will haunt us in our dreams for weeks. Should be filed as a separate issue.